lib/rubocop/cop/lint/ambiguous_block_association.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for ambiguous block association with method
+      # when param passed without parentheses.
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #   # It's ambiguous because there is no parentheses around `a` param
+      #   some_method a { |val| puts val }
+      #
+      # @example
+      #
+      #   # good
+      #
+      #   # With parentheses, there's no ambiguity.
+      #   some_method(a) { |val| puts val }
+      class AmbiguousBlockAssociation < Cop
+        MSG = 'Parenthesize the param to make sure that block will be '\
```

Comment:
Perhaps it'd be better to mention the param name and the method name explicitly in the message.